.github/workflows/ci.yml

```diff
@@ -155,10 +155,21 @@ jobs:
         shell: bash
         run: echo ::set-output name=flavor::$([[ "${{ matrix.os }}" == "cuda" ]] && echo "Cuda" || echo "CPU")
 
+      - name: Set parallelization level
+        id: test-parallel
+        shell: bash
+        run: 
+          if [[ ${{ github.event_name == 'schedule' || (github.event_name == 'push' && (github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/tags/v'))) }} ]]; then
+            echo ::set-output name=testParallel::1
+          else
+            echo ::set-output name=testParallel::4
+          fi
+
       - name: Build and test
         run: dotnet test --configuration=Release --framework=${{ matrix.framework }} -p:TreatWarningsAsErrors=true --logger GitHubActions Src/${{ matrix.library }}.Tests.${{ steps.test-flavor.outputs.flavor }}
         env:
           ILGPU_CLEAN_TESTS: ${{ github.event_name == 'schedule' || (github.event_name == 'push' && (github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/tags/v'))) }}
```

Comment:
The value for this could be set in the same way that we set the parallelisation level

For example you could set both environment variables as a json string when the 'clean tests' conditions are reached, and then use that as the value for `env`. When the conditions are not reached, no values are set, which means we use the default.

Reply:
I didn't get it from first...but please check the latest version when you can, I think i got it 🙏 👀